app/clients/letter/dvla.py

```diff
@@ -45,6 +47,23 @@ class DvlaThrottlingException(DvlaRetryableException):
     pass
 
 
+@contextlib.contextmanager
+def _handle_common_dvla_errors(custom_httperror_exc_handler: Callable[[requests.HTTPError], None] = lambda x: None):
+    try:
+        yield
+    except (ConnectionError, requests.ConnectionError, requests.Timeout) as e:
+        raise DvlaRetryableException() from e
+    except requests.HTTPError as e:
+        custom_httperror_exc_handler(e)
+
+        if e.response.status_code == 429:
+            raise DvlaThrottlingException() from e
+        elif e.response.status_code >= 500:
+            raise DvlaRetryableException() from e
+        else:
+            raise DvlaNonRetryableException() from e
```

Comment:
optional: maybe we can add some context to each of these exceptions just to make them a bit easier to instantly read? I reckon in Sentry we could get the context from `e` and understand it, but having it surfaced immediately might still be nice. Something like (untested...)

```suggestion
            raise DvlaThrottlingException() from e
        elif e.response.status_code >= 500:
            raise DvlaRetryableException(f"Received {e.response.status_code} from {e.request.url}`) from e
        else:
            raise DvlaNonRetryableException(f"Received {e.response.status_code} from {e.request.url}`) from e
```